### PR TITLE
Explicitely drop the `started` var in examples

### LINKED ISSUE #67457: Example CondVar code should explicitly drop the MutexGuard after it's done with it
At the very end of  https://github.com/rust-lang/rust/blob/6b561b4917e803c4be4ca44d8e552b680cb9e380/src/libstd/sync/condvar.rs#L89-L110 and in all similar examples elsewhere in the file, there should be a `std::mem::drop(started)` call. Leaving a condition variable's mutex held after we're no longer watching it will block and possibly deadlock notifier threads.

### PR DESCRIPTION
**Pull request by poliorcetics**
Fixes #67457.

I added the explicit drop with an explanation, it is enough ?

### CODE REVIEW

File: src/libstd/sync/condvar.rs [hunk 1]
```diff
@@ -60,6 +60,10 @@ impl WaitTimeoutResult {
     ///         break
     ///     }
     /// }
+    /// // Leaving a condition variable's mutex held after we're no longer
+    /// // watching it will block and possibly deadlock notifier threads, so
```
Comment (dtolnay) [inline src/libstd/sync/condvar.rs:64]: I think this is misleading. Just holding the mutex would not be sufficient to block a notifier thread. It just stops a notifier thread (or anything else) from acquiring the same mutex, which is the whole point of a mutex. If I understand correctly, the `notify_one` call does not block, so there is nothing about the behavior that would be specific to a notifier thread.

File: src/libstd/sync/condvar.rs [hunk 2]
```diff
@@ -60,6 +60,10 @@ impl WaitTimeoutResult {
...
+    /// // we explicitely drop it.
```
Comment (dtolnay) [inline src/libstd/sync/condvar.rs:65]: The salient point is to release the mutex as soon as possible, not to release it explicitly, so I also find this part misleading. Implicit release when it goes out of scope is just as good as long as it happens as soon as possible.

**Review by dtolnay — CHANGES_REQUESTED** [review]

File: src/libstd/sync/condvar.rs (hunk 2 above)
Comment (poliorcetics) [reply src/libstd/sync/condvar.rs:65]: Then should I just make the comment something like `It is sometimes necessary to manually drop the mutex to unlock it as soon as possible. If you need the ressource until the end of the scope, this is not necessary.` and put it once in a new example on the mutex type itself ?

Comment (dtolnay) [reply src/libstd/sync/condvar.rs:65]: Yeah I think Mutex could use some example code that demonstrates a realistic use case requiring explicit drop.

**Review by dtolnay — COMMENTED** [review]
Per https://github.com/rust-lang/rust/pull/73074#discussion_r436399462 I'll close this and we can follow up in a separate PR adding that example to Mutex.

**Pull request closed without merging**